include/led.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Driver specific representation of a LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable time period (in ms) supported by the device */
+	u32_t min_period;
+	/* Maximum acceptable time period (in ms) supported by the device */
+	u32_t max_period;
+	/* Minimum acceptable brightness (in percent) supported by the device */
```

Comment:
If this is a percentage, why is anything other than 0 to 100 chosen? The device should be able to support 0 if it can turn off, and 100 should just be the maximum value, no?

I'm a little confused about why a percentage is being used, also -- could you clarify?

Since many commonly used digitally addressable LED channels support 256+ brightness levels, using percentage unnecessarily throws away resolution that the hardware can support. Why not at least use a u8_t, specify min and max values of 0 and 0xff in the driver API, and force the LED implementations to ignore low order bits they don't support?

That would let the users of this LED API make uniform calls, with the implementations "doing their best" in the background.

Reply:
Actually, the min and max percent are modelled based on the hardware supported brightness levels. For instance, in the LP3943 driver I have following code to interpret the percentage,
```
val = (value * 255) / dev_data->max_brightness;
```
The driver should take care of interpreting the brightness level based on the underlying hardware. But if this seems too confusing (more it looks like) we can switch to specifying brightness levels in `u8_t`.

Reply:
My point here is that using a percentage means only values from 0 to 100 are supported: less than 7 bits of resolution. This is a problem since a lot of hardware supports at least 8 bits of resolution. Thanks for agreeing to the change.